framework/include/utils/InputParametersChecksUtils.h

```diff
@@ -0,0 +1,549 @@
+  void checkVectorParamsSameLengthIfSet(const std::string & param1,
+                                        const std::string & param2,
+                                        const bool ignore_empty_default_param2 = false) const;
+  /// Check that a vector parameter is the same length as two others combined
+  /// @param param1 vector parameter that provides the target size
+  /// @param param2 vector parameter that provides one term in the combined size
+  /// @param param3 vector parameter that provides one term in the combined size
+  template <typename T, typename S, typename U>
+  void checkVectorParamLengthSameAsCombinedOthers(const std::string & param1,
+                                                  const std::string & param2,
+                                                  const std::string & param3) const;
+
+  /// Check if the user commited errors during the definition of block-wise parameters
+  /// @param block_param_name the name of the parameter that provides the groups of blocks
+  /// @param parameter_names vector of the names of the parameters that are defined on a per-block basis
+  template <typename T>
+  void checkBlockwiseConsistency(const std::string & block_param_name,
+                                 const std::vector<std::string> & parameter_names) const;
+  /// Return whether two parameters are consistent
+  /// @param other_param InputParameters object from another object to check the 'param_name' parameter in
+  /// @param param_name the name of the parameter to check for consistency
+  template <typename T>
+  bool parameterConsistent(const InputParameters & other_param,
+                           const std::string & param_name) const;
+  /// Emits a warning if two parameters are not equal to each other
+  /// @param other_param InputParameters object from another object to check the 'param_name' parameter in
+  /// @param param_name the name of the parameter to check for consistency
+  template <typename T>
+  void warnInconsistent(const InputParameters & parameters, const std::string & param_name) const;
+  /// Error messages for parameters that should depend on another parameter
+  /// @param param1 the parameter has not been set to the desired value (for logging purposes)
+  /// @param value_not_set the desired value (for logging purposes)
+  /// @param dependent_params all the parameters that should not have been since 'param1' was not set to 'value_not_set'
+  void errorDependentParameter(const std::string & param1,
+                               const std::string & value_not_set,
+                               const std::vector<std::string> & dependent_params) const;
+
+private:
+  // Convenience routines so that defining new checks feels very similar to coding checks in
+  // MooseObjects and Actions (MooseParameterInterface-derived classes)
+  /// Forwards parameter check to the class using this utility
+  template <typename T>
+  T forwardGetParam(const std::string & param_name) const
```

Comment:
do you like these or do you think it s obnoxious and I should just have the `_customer_class->template getParam<T>(param_name);` directly in there

Reply:
I know it's ugly when you have to do `this->template blah<T>()`, but what about ones like `forwardIsParamValid`: what's the reason for those?

Reply:
uniformity?

Reply:
Could these maybe go into `MooseBaseParameterInterface`?

Reply:
these few routines forward to routines in `MooseBaseParameterInterface`. we dont particularly need them outside of this parameter checking class, which is why they are `private`